src/main/scala/com/microsoft/hyperspace/index/sources/iceberg/IcebergRelation.scala

```diff
@@ -47,11 +46,11 @@ class IcebergRelation(spark: SparkSession, override val plan: DataSourceV2Relati
   /**
    * Computes the signature of the current relation.
    */
-  override def signature: String = plan.source match {
+  override def signature(fb: FingerprintBuilder): Option[Fingerprint] = plan.source match {
     case _: IcebergSource =>
       val table = loadIcebergTable
       val snapshotId = plan.options.getOrElse("snapshot-id", table.currentSnapshot().snapshotId())
-      snapshotId + table.location()
+      Some(fb.add(snapshotId.toString).add(table.location).build())
```

Comment:
Ditto - if only `Some` is returned then let's drop it and return the `Fingerprint`.

Reply:
Ditto